pennylane/hf/integrals.py

```diff
@@ -308,6 +308,207 @@ def overlap_integral(*args):
+
+    Args:
+        li (integer): angular momentum for the left Gaussian function
+        lj (integer): angular momentum for the right Gaussian function
+        ri (float): position of the first Gaussian function
+        rj (float): position of the second Gaussian function
+        alpha (array[float]): exponent of the first Gaussian function
+        beta (array[float]): exponent of the second Gaussian function
+        e (integer): exponent of the position component
+        r (array[float]): distance between the center of the Hermite Gaussian function and origin
+
+    Returns:
+        array[float]: one-dimensional multipole moment integral between primitive Gaussian functions
+
+    **Example**
+
+    >>> li, lj = 0, 0
+    >>> ri, rj = np.array([2.0]), np.array([2.0])
+    >>> alpha = np.array([3.42525091])
+    >>> beta = np.array([3.42525091])
+    >>> e = 1
+    >>> r = 1.5
+    >>> gaussian_moment(li, lj, ri, rj, alpha, beta, e, r)
+    array([1.0157925])
+    """
+    s = 0.0
+    for t in range(min(li + lj + 1, e + 1)):
+        s = s + expansion(li, lj, ri, rj, alpha, beta, t) * hermite_moment(alpha, beta, t, e, r)
+
+    return s
+
+
+def moment_integral(basis_a, basis_b, e, idx):
+    r"""Return a function that computes the multipole moment integral for two contracted Gaussians.
+
+    The multipole moment integral for two primitive Gaussian functions is computed as
+
+    .. math::
+
+        S^e = \left \langle G_i | q_C^e | G_j \right \rangle
+                   \left \langle G_k | G_l \right \rangle
+                   \left \langle G_m | G_n \right \rangle,
+
+    where :math:`G_{i-n}` is a one-dimensional Gaussian function, :math:`q = x, y, z` is the
+    coordinate at which the integral is evaluated, :math:`C` is the origin of the Cartesian
+    coordinates and :math:`e` is a positive integer. For contracted Gaussians, such
+    integrals will be computed over primitive Gaussians, multiplied by the normalized contraction
+    coefficients and finally summed over.
+
+    The ``idx`` argument determines the coordinate :math:`q` at which the integral is computed. It
+    can be :math:`0, 1, 2` for :math:`x, y, z` components, respectively.
+
+    Args:
+        basis_a (BasisFunction): first basis function
+        basis_b (BasisFunction): second basis function
+        e (integer): exponent of the position component
+        idx (integer): index determining the dimension of the multipole moment integral
+
+    Returns:
+        function: function that computes the multipole moment integral
+
+    **Example**
+
+    >>> symbols  = ['H', 'Li']
+    >>> geometry = np.array([[0.0, 0.0, 0.0], [2.0, 0.0, 0.0]], requires_grad = False)
+    >>> mol = qml.hf.Molecule(symbols, geometry)
+    >>> args = []
+    >>> e, idx =  1, 0
+    >>> moment_integral(mol.basis_set[0], mol.basis_set[1], e, idx)(*args)
+    3.12846324e-01
+    """
+
+    def _moment_integral(*args):
```

Comment:
The arguments that are passed here are used in the private function `_generate_params`. Since the documentation for this function is not available and used "under the hood" it might be useful to provide, in the description of `moment_integral` an example or description of what `*args` can be, and how it is expected to be formatted.